### PR TITLE
OCPBUGS#1823: Default pids-limit and log-max-size updated

### PR DESCRIPTION
**Pull request by amolnar-gh**
Version(s): 4.11, 4.12

Issue: https://issues.redhat.com/browse/OCPBUGS-1823

Link to docs preview: https://52356--docspreview.netlify.app/openshift-enterprise/latest/post_installation_configuration/machine-configuration-tasks.html#create-a-containerruntimeconfig_post-install-machine-configuration-tasks

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/create-a-containerruntimeconfig-crd.adoc [hunk 1]
```diff
@@ -15,11 +15,10 @@ To revert the changes implemented by using a `ContainerRuntimeConfig` CR, you mu
 
 You can modify the following settings by using a `ContainerRuntimeConfig` CR:
 
-* **PIDs limit**: The `pidsLimit` parameter sets the CRI-O `pids_limit` parameter, which is maximum number of processes allowed in a container. The default is 1024 (`pids_limit = 1024`).
+* **PIDs limit**: The `--pod-pids-limit` Kubelet flag sets the CRI-O `pids_limit` parameter, which is maximum number of processes allowed in a container. The default is 0 (`pids_limit = 0`).
```
Comment (haircommander) [inline modules/create-a-containerruntimeconfig-crd.adoc:18]: this was deprecated upstream, but not pulled down downstream. The trouble has been dropping support in the container runtime config crd.

I would reword this as:
```
Setting PIDs limit in the `ContainerRuntimeConfig` is in the process of being deprecated. If PIDs limits are required, the recommendation is to use the `KubeletConfig` field `PodPidsLimit` instead. Note: the CRI-O flag is applied on the container's cgroup, whereas the Kubelet's is set on the pod's cgroup. Please adjust the PIDs limit accordingly.
```

File: modules/create-a-containerruntimeconfig-crd.adoc [hunk 2]
```diff
@@ -15,11 +15,10 @@ To revert the changes implemented by using a `ContainerRuntimeConfig` CR, you mu
...
 * **Overlay size**: The `overlaySize` parameter sets the CRI-O Overlay storage driver `size` parameter, which is the maximum size of a container image.
-* **Maximum log size**: The `logSizeMax` parameter sets the CRI-O `log_size_max` parameter, which is the maximum size allowed for the container log file. The default is unlimited (`log_size_max = -1`). If set to a positive number, it must be at least 8192 to not be smaller than  the ConMon read buffer. ConMon is a program that
-monitors communications between a container manager, such as Podman or CRI-O, and the OCI runtime, such as runC or crun, for a single container.
+* **Maximum log size**: The `--container-log-max-size` Kubelet flag sets the CRI-O `log_size_max` parameter, which is the maximum size allowed for the container log file. The default is unlimited (`log_size_max = -1`). If set to a positive number, it must be at least 8192 to not be smaller than  the ConMon read buffer. ConMon is a program that monitors communications between a container manager, such as Podman or CRI-O, and the OCI runtime, such as runC or crun, for a single container.
```
Comment (haircommander) [inline modules/create-a-containerruntimeconfig-crd.adoc:21]: Same here: another suggested reword:

```
Setting maximum log size in the `ContainerRuntimeConfig` is in the process of being deprecated. If a maximum log size is required, the recommendation is to use the `KubeletConfig` field `ContainerLogMaxSize` instead.
```

**Comment by haircommander** [comment]
also related to https://issues.redhat.com/browse/OCPBUGS-3011

File: modules/create-a-containerruntimeconfig-crd.adoc (hunk 2 above)
Comment (amolnar-rh) [reply modules/create-a-containerruntimeconfig-crd.adoc:21]: @haircommander Should I include this sentence as well? 
> If set to a positive number, it must be at least 8192 to not be smaller than  the ConMon read buffer. ConMon is a program that monitors communications between a container manager, such as Podman or CRI-O, and the OCI runtime, such as runC or crun, for a single container.

Comment (haircommander) [reply modules/create-a-containerruntimeconfig-crd.adoc]: I don't believe the kubelet is limited in the same way

Comment (amolnar-rh) [reply modules/create-a-containerruntimeconfig-crd.adoc]: Removed it then, thanks

**Review by lpettyjo — COMMENTED** [review]
You have two commits. This needs to be squashed to one commit. Otherwise, LGTM.

**Comment by umohnani8** [comment]
LGTM

**Comment by snarayan-redhat** [comment]
@amolnar-rh - 4.11 needs a manual CP. Let me know once it's created and I can merge it for you. 4.12 is merged.
